server/src/test/java/org/elasticsearch/transport/InboundPipelineTests.java

```diff
@@ -54,15 +54,18 @@ public void testPipelineHandling() throws IOException {
                 final Version version = header.getVersion();
                 final boolean isRequest = header.isRequest();
                 final long requestId = header.getRequestId();
-                final boolean isCompressed = header.isCompressed();
+                final Compression.Scheme compressionScheme = header.getCompressionScheme();
+                if (header.isCompressed()) {
+                    assertNotNull(compressionScheme);
+                }
```

Comment:
Can we add `else assertNull(compressionScheme);` ? (or embed it into the other assertion).